Flick/Cells/NotificationTableViewCell.swift

```diff
@@ -0,0 +1,170 @@
+        }
+
+        profileImageView.snp.makeConstraints { make in
+            make.top.bottom.leading.equalTo(containerView).inset(padding)
+            make.height.width.equalTo(40)
+        }
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    private func layoutNotificationLabel() {
+        notificationLabel.snp.makeConstraints { make in
+            make.centerY.equalTo(profileImageView)
+            make.leading.equalTo(profileImageView.snp.trailing).offset(padding)
+            make.trailing.equalTo(containerView).inset(12)
+        }
+    }
+
+    private func setupFriendRequestCell(fromUser: String) {
+        let friendLabelString = NSMutableAttributedString(string: fromUser, attributes: [NSAttributedString.Key.font: UIFont.boldSystemFont(ofSize: 14)])
+        let friendRequestString = NSMutableAttributedString(string: " sent you a friend request")
+        friendLabelString.append(friendRequestString)
+        notificationLabel.attributedText = friendLabelString
+
+        acceptButton.setTitle("Accept", for: .normal)
+        acceptButton.titleLabel?.font = .systemFont(ofSize: 14)
+        acceptButton.layer.backgroundColor = UIColor.lightPurple.cgColor
+        acceptButton.setTitleColor(.gradientPurple, for: .normal)
+        acceptButton.layer.cornerRadius = 17
+        contentView.addSubview(acceptButton)
+
+        ignoreButton.setTitle("Ignore", for: .normal)
+        ignoreButton.titleLabel?.font = .systemFont(ofSize: 14)
+        ignoreButton.layer.backgroundColor = UIColor.lightGray2.cgColor
+        ignoreButton.setTitleColor(.darkBlueGray2, for: .normal)
+        ignoreButton.layer.cornerRadius = 17
+        contentView.addSubview(ignoreButton)
+
+        layoutNotificationLabel()
+
+        let buttonSize = CGSize(width: 96, height: 41)
+
+        profileImageView.snp.remakeConstraints { remake in
+            remake.top.leading.equalTo(containerView).inset(padding)
+            remake.height.width.equalTo(40)
+        }
+
+        acceptButton.snp.makeConstraints { make in
+            make.size.equalTo(buttonSize)
+            make.leading.equalTo(notificationLabel)
+            make.top.equalTo(profileImageView.snp.bottom).offset(10)
+            make.bottom.equalToSuperview().inset(padding)
+        }
+
+        ignoreButton.snp.makeConstraints { make in
+            make.size.equalTo(buttonSize)
+            make.leading.equalTo(acceptButton.snp.trailing).offset(48)
+            make.top.bottom.equalTo(acceptButton)
+        }
+
+    }
+
+    private func setupFriendAcceptCell(fromUser: String) {
+        let friendLabelString = NSMutableAttributedString(string: fromUser, attributes: [NSAttributedString.Key.font: UIFont.boldSystemFont(ofSize: 14)])
+        let friendAcceptString = NSMutableAttributedString(string: " accepted your friend request ")
+        friendLabelString.append(friendAcceptString)
```

Comment:
it looks to me that all the setup cell functions have these three line that just differ by the string on the second line. I think it's prob better the move these into a separate function that returns a `NSMutableAttributedString`?